packages/compass-indexes/src/components/indexes-table/indexes-table.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useMemo } from 'react';
+import {
+  css,
+  Table,
+  TableHeader,
+  Row,
+  Cell,
+  cx,
+  spacing,
+} from '@mongodb-js/compass-components';
+
+import NameField from './name-field';
+import TypeField from './type-field';
+import SizeField from './size-field';
+import UsageField from './usage-field';
+import PropertyField from './property-field';
+import DropField from './drop-field';
+
+// When row is hovered, we show the delete button
+const rowStyles = css({
+  ':hover': {
+    '.delete-cell': {
+      visibility: 'visible',
+    },
+  },
+});
+// When row is not hovered, we hide the delete button
+const deletFieldStyles = css({
+  visibility: 'hidden',
+});
+
+const cellStyles = css({
+  verticalAlign: 'top',
+});
+
+const nameFieldStyles = css({
+  paddingTop: spacing[2],
+  paddingBottom: spacing[2],
+});
+
+// todo: move to redux store when converting that to ts
+export type IndexModel = {
+  name: string;
+  fields: {
+    serialize: () => { field: string; value: number | string }[];
+  };
+  type: 'geo' | 'hashed' | 'text' | 'wildcard' | 'clustered' | 'columnstore';
+  cardinality: 'single' | 'compound';
+  properties: ('unique' | 'sparse' | 'partial' | 'ttl' | 'collation')[];
+  extra: Record<string, string | number | Record<string, any>>;
+  size: number;
+  relativeSize: number;
+  usageCount?: number;
+  usageSince?: Date;
+};
+
+type IndexesTableProps = {
+  indexes: IndexModel[];
+  canDeleteIndex: boolean;
+  onSortTable: (name: string, direction: 'asc' | 'desc') => void;
+  onDeleteIndex: (name: string) => void;
+};
+
+export const IndexesTable: React.FunctionComponent<IndexesTableProps> = ({
+  indexes,
+  canDeleteIndex,
+  onSortTable,
+  onDeleteIndex,
+}) => {
+  const columns = useMemo(() => {
+    const _columns = [
+      'Name and Definition',
+      'Type',
+      'Size',
+      'Usage',
+      'Properties',
+    ].map((name) => {
+      return (
+        <TableHeader
```

Comment:
Looking at the designs, it looks like we can remove those vertical borders between the header cells. This might be something we get for free with upgrading leafygreen versions, but for now would it make sense to add:
```ts
const tableHeaderStyles = css({
  borderWidth: '0px 0px 3px 0px',
  '> div': {
    justifyContent: 'start'
  }
});
```
```suggestion
        <TableHeader
          className={tableHeaderStyles}
```

Before/After:
<img width="962" alt="Screen Shot 2022-08-19 at 2 41 32 PM" src="https://user-images.githubusercontent.com/1791149/185708652-58be9f63-db4a-4109-8010-5dc9f494c5e8.png">
<img width="907" alt="Screen Shot 2022-08-19 at 5 10 24 PM" src="https://user-images.githubusercontent.com/1791149/185708654-381b2b7b-d2d4-492e-b9b3-dcfeb5620003.png">
Or would it be better to wait until we update leafygreen versions and avoid hacks here?

Reply:
i did end up removing these borders as the table looks cleaner.